src/rust/inetstack/protocols/tcp/peer.rs

```diff
@@ -303,30 +301,36 @@ impl<const N: usize> TcpPeer<N> {
     ) -> Poll<Result<(QDesc, SocketAddrV4), Fail>> {
         let mut inner: RefMut<Inner<N>> = self.inner.borrow_mut();
 
-        let cb: ControlBlock<N> = match inner.qtable.borrow_mut().get_mut(&qd) {
-            Some(InetQueue::Tcp(queue)) => match queue.get_mut_socket() {
+        let cb: ControlBlock<N> = {
+            let mut qtable: RefMut<IoQueueTable> = inner.qtable.borrow_mut();
+            let queue: &mut TcpQueue<N> = qtable.get_mut::<TcpQueue<N>>(&qd)?;
+            match queue.get_mut_socket() {
                 Socket::Listening(socket) => match socket.poll_accept(ctx) {
                     Poll::Pending => return Poll::Pending,
                     Poll::Ready(result) => match result {
                         Ok(cb) => cb,
                         Err(err) => {
-                            inner.qtable.borrow_mut().free(&new_qd);
+                            inner
+                                .qtable
+                                .borrow_mut()
+                                .free::<TcpQueue<N>>(&new_qd)
+                                .expect("queue should exist");
```

Comment:
To follow out guidelines, could you add a comment here to explain why it is safe to call `expect()` here?